### PR TITLE
schedulers/hot: let leader can be schedule even if the peer is not balanced

### PR DESCRIPTION
**Pull request by nolouch**
…lanced

Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
In this scenario: 9 tikv, running with a workload  `oltp_insert` (6 tables)
- the count is not suitable for calculating limit because there are many hot regions (index) and only 6 very hot regions (row data)
- there are too many hot regions, so the hot leader may have no chance to be scheduled.
image
image

### What is changed and how it works?
- remove the auto limit
- randomly schedule peer or leader.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test

### CODE REVIEW

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -400,14 +404,19 @@ func (h *hotScheduler) balanceHotReadRegions(cluster opt.Cluster) []*operator.Op
 
 func (h *hotScheduler) balanceHotWriteRegions(cluster opt.Cluster) []*operator.Operator {
 	// prefer to balance by peer
-	peerSolver := newBalanceSolver(h, cluster, write, movePeer)
-	ops := peerSolver.solve()
-	if len(ops) > 0 {
-		return ops
+	seed := h.r.Intn(randomUpperBound-randomLowBound) + randomLowBound
```
Comment (rleungx) [inline server/schedulers/hot_region.go]: Why we need these two bound since it cannot be changed?

File: server/schedulers/hot_region.go [hunk 2]
```diff
@@ -153,12 +157,12 @@ func (h *hotScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
 }
 
 func (h *hotScheduler) allowBalanceLeader(cluster opt.Cluster) bool {
-	return h.OpController.OperatorCount(operator.OpHotRegion) < minUint64(h.leaderLimit, cluster.GetHotRegionScheduleLimit()) &&
+	return h.OpController.OperatorCount(operator.OpHotRegion) < cluster.GetHotRegionScheduleLimit() &&
```
Comment (lhy1024) [inline server/schedulers/hot_region.go]: why to remove limit?

Comment (nolouch) [reply server/schedulers/hot_region.go]: The `count` is not suitable for calculating the limit for now and we already have a pending influence.

**Pull request merged**